types/address.go

```diff
@@ -31,6 +31,17 @@ const (
 	Bech32PrefixConsPub = "cosmosvalconspub"
 )
 
+// Address is a common interface for different types of addresses used by the SDK
+type Address interface {
+	Equals([]byte) bool
+	Empty() bool
+	Marshal() ([]byte, error)
+	MarshalJSON() ([]byte, error)
+	Bytes() []byte
+	String() string
+	Format(s fmt.State, verb rune)
+}
+
```

Comment:
Can we assert that the address types we utilize actually implement this interface?

Reply:
second that

Reply:
Pushed commit with test and ☝️

Reply:
Awesome. Correct me if I'm wrong, I think they (also) meant to do `var _ Address = AccAddress{}`, etc...

Reply:
I can do that...